api/v1alpha1/selfnoderemediationconfig_webhook_test.go

```diff
@@ -0,0 +1,141 @@
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+type TestDataItem struct {
+	fieldName     string
+	fieldValue    string
+	fieldMinValue string
+	errorMessage  string
+}
+
+var _ = Describe("SelfNodeRemediationConfig Validation", func() {
+
+	Describe("validate update time fields of SelfNodeRemediationConfig CR", func() {
+		shortTimeTestItems := []TestDataItem{
+			{"PeerApiServerTimeout", "1.2ms", MinDurPeerApiServerTimeout, ErrPeerApiServerTimeout},
+			{"ApiServerTimeout", "1.2ms", MinDurApiServerTimeout, ErrApiServerTimeout},
+			{"PeerDialTimeout", "1.2ms", MinDurPeerDialTimeout, ErrPeerDialTimeout},
+			{"PeerRequestTimeout", "1.2ms", MinDurPeerRequestTimeout, ErrPeerRequestTimeout},
+			{"ApiCheckInterval", "1.2ms", MinDurApiCheckInterval, ErrApiCheckInterval},
+			{"PeerUpdateInterval", "1.2ms", MinDurPeerUpdateInterval, ErrPeerUpdateInterval},
+		}
+
+		for _, item := range shortTimeTestItems {
+			item := item
+			text := "for " + item.fieldName + " value shorter than " + item.fieldMinValue
+			Context(text, func() {
+				It("should be rejected", func() {
+					snrcOld := createDefaultSelfNodeRemediationConfigCR()
+					snrc := createSelfNodeRemediationConfigCR(item.fieldName, item.fieldValue)
+
+					err := snrc.ValidateUpdate(snrcOld)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring(item.errorMessage))
+				})
+			})
+		}
+
+		negativeTimeTestItems := []TestDataItem{
+			{"PeerApiServerTimeout", "-1ms", MinDurPeerApiServerTimeout, ErrPeerApiServerTimeout},
+			{"ApiServerTimeout", "-1ms", MinDurApiServerTimeout, ErrApiServerTimeout},
+			{"PeerDialTimeout", "-1ms", MinDurPeerDialTimeout, ErrPeerDialTimeout},
+			{"PeerRequestTimeout", "-1ms", MinDurPeerRequestTimeout, ErrPeerRequestTimeout},
+			{"ApiCheckInterval", "-1ms", MinDurApiCheckInterval, ErrApiCheckInterval},
+			{"PeerUpdateInterval", "-1ms", MinDurPeerUpdateInterval, ErrPeerUpdateInterval},
+		}
+
+		for _, item := range negativeTimeTestItems {
+			item := item
+			text := "for " + item.fieldName + " with negative value" + item.fieldMinValue
+			Context(text, func() {
+				It("should be rejected", func() {
+					snrcOld := createDefaultSelfNodeRemediationConfigCR()
+					snrc := createSelfNodeRemediationConfigCR(item.fieldName, item.fieldValue)
+					err := snrc.ValidateUpdate(snrcOld)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring(item.errorMessage))
+				})
+			})
+		}
+
+		Context("for valid CR", func() {
+			It("should not be rejected", func() {
+				snrcOld := createDefaultSelfNodeRemediationConfigCR()
+				snrc := createDefaultSelfNodeRemediationConfigCR()
```

Comment:
Nit: I think it's better to use more diverse value set - at least some values should be different from original values